src/prefect/utilities/asyncutils.py

```diff
@@ -115,7 +115,10 @@ async def context_local_wrapper():
         """
         token = RUN_ASYNC_FLAG.set(True)
         try:
-            result = await coroutine
+            if inspect.iscoroutine(coroutine):
```

Comment:
was getting `cannot await function`

Reply:
Where were you seeing this error? There might be something that shouldn't be decorated with `@sync_compatible`

Reply:
Can you remove this change? I'd like to investigate this further because we shouldn't be passing non-coroutines to `run_sync`.

Reply:
yep!